### PR TITLE
add new module for Webapp slot

### PR DESCRIPTION
**Pull request by yungezz**

### CODE REVIEW

File: library/azure_rm_webapp_slot.py [hunk 1]
```diff
@@ -0,0 +1,993 @@
...
+    location:
+        description:
+            - Resource location. If not set, location from the resource group will be used as default.
+    configuration_source:
+        description:
+            - Source slot to clone configurations from. Use webapp's name to refer to the production slot.
+    auto_swap_slot_name:
+        description:
+            - Target slot name to auto swap.
+            - Set it to 'None' to disable auto slot swap.
```
Comment (zikalino) [inline library/azure_rm_webapp_slot.py]: What does it mean set it to "None"? should be string?

I think we should avoid using "none"

File: library/azure_rm_webapp_slot.py [hunk 2]
```diff
@@ -0,0 +1,993 @@
...
+          - name: "java"
+            version: "8"
+            settings:
+              java_container: "Tomcat"
+              java_container_version: "8.5"
+'''
+
+RETURN = '''
+azure_webapp:
+    description: Id of current web app.
```
Comment (zikalino) [inline library/azure_rm_webapp_slot.py]: shouldn't "id" be just on the top?

i mean "azure_webapp" is a dict, and description is "Id of current web app."

Comment (zikalino) [reply library/azure_rm_webapp_slot.py]: self.results contains "id", so there's no "azure_webapp" dict

Comment (yungezz) [reply library/azure_rm_webapp_slot.py]: fixed

File: library/azure_rm_webappslot.py [hunk 3]
```diff
@@ -42,18 +42,19 @@
...
+            - Set it target slot name to auto swap.
+            - Set it to False to disable auto slot swap.
     swap:
         description:
             - Swap deployment slots of a web app.
         suboptions:
             action:
                 description:
                     - Swap types.
-                    - 'preview' is to apply target slot's settings on source source slot first.
```
Comment (zikalino) [inline library/azure_rm_webappslot.py]: Maybe use C(preview) etc?

File: library/azure_rm_webappslot.py [hunk 4]
```diff
@@ -120,6 +120,10 @@
             registry_server_password:
                 description:
                     - The container registry server password.
+    startup_file:
+        description:
+            - The slot startup file.
+            - This only applies for linux web app slot.
```
Comment (zikalino) [inline library/azure_rm_webappslot.py]: This only applies to...

File: library/azure_rm_webappslot.py [hunk 5]
```diff
@@ -120,6 +120,10 @@
...
     app_settings:
         description:
             - Configure web app slot application settings. Suboptions are in key value pair format.
```
Comment (zikalino) [inline library/azure_rm_webappslot.py]: so it's a dict, right?
Maybe should be "A dictionary containing Web App slot settings"?

**Comment by Fred-sun** [comment]
@yungezz

**Comment by Fred-sun** [comment]
@yungezz need you update

**Comment by Fred-sun** [comment]
@yungezz

**Comment by Fred-sun** [comment]
@yungezz

**Comment by Fred-sun** [comment]
@yungezz

**Pull request closed without merging**